### ISSUE TITLE
implement challenge article function

### ISSUE THREAD

**Issue by ringaile** [ringaile/honest_media#10]
we can simply do something like check a check-box to do it.

Once an article is triggered, we should do:

- who has to prove it?
- etc...

**Comment by ajmachado** [comment]
Backend: I have added challenge struct and mapping in the reader role contract. Do you think we need to have a separate contract for it like the articles?

**Comment by ajmachado** [comment]
Frontend: When the reader clicks on 'challenge' (Check-box or button), they will be shown a file upload button(to upload proof) and a text box to enter the amount they are staking on it.

**Comment by ringaile** [comment]
_Do you think we need to have a separate contract for it like the articles?_
i guess, not yet. at least i do not think so :)

_Frontend: When the reader clicks on 'challenge' (Check-box or button), they will be shown a file upload button(to upload proof) and a text box to enter the amount they are staking on it._

and then the validators have to review and prove it?
uploading the proof should be difficult. we can focus for now, to include some text (like urls for example). And then a reader submits a proof, for example, three randomly selected validators should read it and approve it or not. once it is approved by three validators, the stake is distributed. what do you think?

we really need to think about how to break it down to small steps so it is easier to implement it :)

**Comment by ajmachado** [comment]
_uploading the proof should be difficult. we can focus for now, to include some text (like urls for example)._ 
It will just be like uploading the article and reference file. If we go with a text box, we will have to save it in a file before uploading it to IPFS anyways.

_And then a reader submits a proof, for example, three randomly selected validators should read it and approve it or not. once it is approved by three validators, the stake is distributed._ 
Yes, we can do 2 of 3 consensus to check if the challenge is successful and then distribute the stake between the validators and the contributor

**Comment by ajmachado** [comment]
Also, we should not allow challenges after one month of the article being published. What do you think?

**Comment by ringaile** [comment]
For now, let’s keep all time constraints as a nice to have and we implement only if we have enough time

**Comment by ajmachado** [comment]
ok

**Comment by ringaile** [comment]
is this frontend too?

**Comment by ajmachado** [comment]
Let's keep this as backend and separate the frontend.

**Comment by ajmachado** [comment]
How should we distribute the amount staked between Reader and validators if challenge won? and between contributor and validator if challenge lost?

**Comment by ringaile** [comment]
i guess, in equal parts as you mentioned:
if reader wins -> reader and validators
if contributor wins -> contributor and validators

**Comment by ajmachado** [comment]
We can close this too

**Comment by ajmachado** [comment]
Just wanted to check if the challenge front end and backend are linked.

**Comment by ajmachado** [comment]
The 'Challenge' HTML elements need to be moved to the next line and a text field is needed for the Reader's address. We also need a file upload button for the reader to upload 'proof' for the challenge.
